test/jdk/java/nio/channels/DatagramChannel/SelectWhenRefused.java

```diff
@@ -68,7 +71,10 @@ public static void main(String[] args) throws IOException {
                 if (n > 0) {
                     sel.selectedKeys().clear();
                     try {
-                        n = dc.read(ByteBuffer.allocate(100));
+                        ByteBuffer buf = ByteBuffer.allocate(100);
+                        n = dc.read(buf);
+                        String message = new String(buf.array());
+                        System.out.format("received %s at %s%n", message, dc.getLocalAddress());
```

Comment:
The intermittent failure is seen on the previous test scenario, so some diagnotics relating to that are of interest
I think we are interested in the sender detail here ... where did the stray datagram come from, which would necessitate invoking receive on the datagram channel ... but as of yet there has been no failures observed here afaik

Also prior to each DC send it could be worth invoking selectNow on the selector to disperse any stray pending events on the low level OS "select"